features/setup.py

```diff
@@ -1,82 +1,39 @@
 #!/usr/bin/env python
+###########################
+# This code block is a HACK (!), but is necessary to avoid code duplication. Do NOT alter these lines.
 import os
-import shutil
-import subprocess
-
-from setuptools import setup, find_packages
-
-cwd = os.path.dirname(os.path.abspath(__file__))
-
-with open(os.path.join(os.path.dirname(__file__), '..', 'VERSION')) as version_file:
-    version = version_file.read().strip()
-
-"""
-To release a new stable version on PyPi, simply tag the release on github, and the Github CI will automatically publish
-a new stable version to PyPi using the configurations in .github/workflows/pypi_release.yml .
-You need to increase the version number after stable release, so that the nightly pypi can work properly.
-"""
-try:
-    if not os.getenv('RELEASE'):
-        from datetime import date
-        minor_version_file_path = os.path.join('..', 'VERSION.minor')
-        if os.path.isfile(minor_version_file_path):
-            with open(minor_version_file_path) as f:
-                day = f.read().strip()
-        else:
-            today = date.today()
-            day = today.strftime("b%Y%m%d")
-        version += day
-except Exception:
-    pass
-
-def create_version_file():
-    global version, cwd
-    print('-- Building version ' + version)
-    version_path = os.path.join(cwd, 'src', 'autogluon', 'features', 'version.py')
-    with open(version_path, 'w') as f:
-        f.write('"""This is autogluon version file."""\n')
-        f.write("__version__ = '{}'\n".format(version))
-
-
-long_description = open(os.path.join('..', 'README.md')).read()
-
-python_requires = '>=3.6, <3.8'
-
+from setuptools import setup
+import importlib.util
+filepath = os.path.abspath(os.path.dirname(__file__))
+filepath_import = os.path.join(filepath, '..', 'core', 'src', 'autogluon', 'core', '_setup_utils.py')
+spec = importlib.util.spec_from_file_location("ag_min_dependencies", filepath_import)
+ag = importlib.util.module_from_spec(spec)
+# Identical to `from autogluon.core import _setup_utils as ag`, but works without `autogluon.core` being installed.
+spec.loader.exec_module(ag)
+###########################
+
+version = ag.load_version_file()
+version = ag.update_version(version)
+
+submodule = 'features'
 requirements = [
-    'numpy==1.19.5',
-    'pandas>=1.0.0,<2.0',
-    'scikit-learn>=0.22.0,<0.25',
-    f'autogluon.core=={version}'
+    'numpy',
```

Comment:
nit: to avoid confusion, please add a comment that pinned versions will be added in `get_dependency_version_ranges` call

Reply:
added